### PR TITLE
Fixes #14825: Don't require organization to search repositories

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by daviddavis** [comment]
What about [the api doc](https://git.io/vw2AQ)?

**Comment by ehelms** [comment]
@daviddavis good catch, updated

**Comment by daviddavis** [comment]
This works well when I supply a product_id or organization_id. However, if I forget to supply one, I get: `undefined method 'default_content_view' for nil:NilClass`.

**Comment by ehelms** [comment]
Updated -- now it will check if @organization has been set and if not will default to displaying for all organizations.

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -105,7 +105,13 @@ def index_relation_environment(query)
         instance_ids += instances.where(:library_instance_id => nil)
         query = Repository.where(:id => instance_ids)
       elsif (params[:library] && !params[:environment_id]) || (params[:environment_id].blank? && params[:content_view_version_id].blank? && params[:content_view_id].blank?)
-        query = query.where(:content_view_version_id => @organization.default_content_view.versions.first.id)
+
+        if @organization
+          query = query.where(:content_view_version_id => @organization.default_content_view.versions.first.id)
+        else
+          content_view_version_ids = Organization.all.map(&:default_content_view).map(&:versions).map(&:first).map(&:id)
+          query = query.where(:content_view_version_id => content_view_version_ids)
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/repositories_controller.rb]: I think you can probably just do this:

```
query = query.in_default_view
```

**Comment by ehelms** [comment]
Good catch -- updated

**Comment by daviddavis** [comment]
ACK

**Pull request merged**